test/Constraints/protocols.swift

```diff
@@ -216,6 +217,12 @@ func staticExistential(_ p: P.Type, pp: P.Protocol) {
   // Static member of metatype -- not allowed
   _ = pp.tum // expected-error{{static member 'tum' cannot be used on protocol metatype 'P.Protocol'}}
   _ = P.tum // expected-error{{static member 'tum' cannot be used on protocol metatype 'P.Protocol'}}
+
+  // Concrete associated type access through protocol metatype
+  _ = pp.E.self
```

Comment:
Nit: In this case `E` is simply a `typealias`, a concrete associated type is an actual associated type that has been same-type constrained.

Reply:
I think I might have been a bit too tired when I wrote this comment, nice spot.